src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java

```diff
@@ -0,0 +1,1322 @@
+      return this.getUseNetwork();
+    }
+    else if (SETTING_NETWORK.equals(attribute)) {
+      return this.getNetwork();
+    }
+    else if (SETTING_USE_USB.equals(attribute)) {
+      return this.getUseUsb();
+    }
+    else if (SETTING_USB_DEVICE.equals(attribute)) {
+      return this.getUsbDevice();
+    }
+    else if (SETTING_MAX_VECTOR_CUT_SPEED.equals(attribute)) {
+      return this.getMaxVectorCutSpeed();
+    }
+    else if (SETTING_MAX_VECTOR_MOVE_SPEED.equals(attribute)) {
+      return this.getMaxVectorMoveSpeed();
+    }
+    else if (SETTING_MIN_POWER.equals(attribute)) {
+      return this.getLaserPowerMin();
+    }
+    else if (SETTING_MAX_POWER.equals(attribute)) {
+      return this.getLaserPowerMax();
+    }
+    else if (SETTING_BED_WIDTH.equals(attribute)) {
+      return this.getBedWidth();
+    }
+    else if (SETTING_BED_HEIGHT.equals(attribute)) {
+      return this.getBedHeight();
+    }
+    else if (SETTING_USE_BIDIRECTIONAL_RASTERING.equals(attribute)) {
+      return this.getUseBidirectionalRastering();
+    }
+    return null;
+  }
+
+  @Override
+  public void setProperty(String attribute, Object value) {
+    if (SETTING_USE_FILE.equals(attribute)) {
+      this.setUseFilename((Boolean) value);
+    }
+    else if (SETTING_FILE.equals(attribute)) {
+      this.setFilename((String) value);
+    }
+    else if (SETTING_USE_NETWORK.equals(attribute)) {
+      this.setUseNetwork((Boolean) value);
+    }
+    else if (SETTING_NETWORK.equals(attribute)) {
+      this.setNetwork((String) value);
+    }
+    else if (SETTING_USE_USB.equals(attribute)) {
+      this.setUseUsb((Boolean) value);
+    }
+    else if (SETTING_USB_DEVICE.equals(attribute)) {
+      this.setUsbDevice((String) value);
+    }
+    else if (SETTING_MAX_VECTOR_CUT_SPEED.equals(attribute)) {
+      this.setMaxVectorCutSpeed((Integer)value);
+    }
+    else if (SETTING_MAX_VECTOR_MOVE_SPEED.equals(attribute)) {
+      this.setMaxVectorMoveSpeed((Integer)value);
+    }
+    else if (SETTING_MIN_POWER.equals(attribute)) {
+      try {
+        this.setLaserPowerMin((Integer)value);
+      }
+      catch (Exception e) {
+        this.setLaserPowerMin(Integer.parseInt((String)value));
+      }
+    }
+    else if (SETTING_MAX_POWER.equals(attribute)) {
+      this.setLaserPowerMax((Integer)value);
+    }
+    else if (SETTING_BED_HEIGHT.equals(attribute)) {
+      this.setBedHeigth((Double)value);
+    }
+    else if (SETTING_BED_WIDTH.equals(attribute)) {
+      this.setBedWidth((Double)value);
+    }
+    else if (SETTING_USE_BIDIRECTIONAL_RASTERING.equals(attribute)) {
+      this.setUseBidirectionalRastering((Boolean) value);
+    }
+  }
+
+  @Override
+  public String[] getPropertyKeys() {
+    return settingAttributes;
+  }
+
+}
+
+
+/* =================================================================================== */
+
+class ByteStream
+{
+  private static final byte[] encode_table = {
```

Comment:
Hard coding this table makes it hard to swap it out if other magic numbers are needed.

Reply:
fixed

Reply:
still missing: configuration option for other magic numbers